src/libstore/outputs-spec.cc

```diff
@@ -0,0 +1,159 @@
+#include "outputs-spec.hh"
+#include "store-api.hh"
+#include "nlohmann/json.hpp"
+
+#include <regex>
+
+namespace nix {
+
+bool OutputsSpec::contains(const std::string & outputName) const
```

Comment:
```suggestion
bool OutputsSpec::wants(const std::string & outputName) const
```

`contains` suggests that something exists, which we may not know yet. `wants` does not suggest this.

Reply:
I have been thinking of `OutputSpec` as a merely the "finite power set of string ∪ ⊤", and so "contains" meant set contains for this notion of set of strings.

It is true that `OutputSpec` is usually *used* to represent the set of wanted outputs, but I feel a little odd calling the methods because I don't think it is the `OutputSpec` *itself* that does the wanting, it is merely used as means to convey something else's wants.